### ISSUE TITLE
Dynamic access permission at submission level

### ISSUE THREAD

**Issue by abhilash-aot** [formio/formio#1073]
Is there any way to **dynamically manage access permission at a submission level rather than at a form level.** 
 
**Use-case:**

Our system uses form.io with two predefined roles: Reviewer and Client and creates a form say **"General Business Application"**.

When the Client fills-in and submit the form; it goes to the "Reviewer" queue for approval. As per the current flow, the client is allowed to update the submission even if it's in the review state. We would need to have the feasibility to  restrict/enable the client to make any changes to the submission made.

**Implementation details are as follows,**

1. Roles defined in formio: Reviewer and Client.
2. At the time of creating a new form, say **"General Business Application"**, we define submissionAccess with **"create_own/update_own/read_own"** to role "Client". So, in effect, all Submissions of the form will have this pre-defined submission access.
3. When a user "User1" with the "Client" role makes a submission for **"General Business Application"**; it will create a new submission **"submission1"**.

Here, I look forward to tweaking the submissionAccess of submission **"submission1"** as listed (not at Form level),

1.  "read_own" for "User1"  with the "Client" role 
2.  "update_own" & "read_own"  for "User2" with the  role "Reviewer"

In short, we want to dynamically control the submissionAccess at submission level. Please guide us in the right direction for achieving this.

**Comment by randallknutson** [comment]
You can do this with "Field Based Resource Access" settings. Basically how this works is you can have a select component where you can assign a user on the submission and grant access to the submission based on that field. For example, if you create a job for an employee and that employee is listed on the job you can grant access because they are listed. You can see more here: https://help.form.io/userguide/roles-and-permissions/#field-based-resource-access

**Comment by ZenMasterJacob20011** [comment]
Closing this thread as it is outdated. Please re-open if it is still relevant. Thank you for your contribution!
